cmake/thirdParties.cmake

```
@@ -1,23 +1,67 @@
 if(MATIO_WITH_HDF5)
-    find_package(HDF5 1.8)
+    if(NOT DEFINED HDF5_USE_STATIC_LIBRARIES)
+        if(MATIO_SHARED)
+            set(HDF5_USE_STATIC_LIBRARIES FALSE)
```

```suggestion
            set(HDF5_USE_STATIC_LIBRARIES FALSE)
            add_definitions(-DH5_BUILT_AS_DYNAMIC_LIB)
```

I only get it correctly linked with shared hdf5.lib on VS if H5_BUILT_AS_DYNAMIC_LIB is defined when building matio.

---

Hm, still struggling.

---

Looks like I made a mistake and it now always links with `hdf5::hdf5-static`. The `hdf5::hdf5-shared` target should automatically define `H5_BUILT_AS_DYNAMIC_LIB`.

What HDF5 version do you use, because I cannot find `H5_BUILT_AS_DYNAMIC_LIB` in the HDF5 1.8 CMake config file, so we have do define it for this version ourselves.

---

I use hdf5/1.12.0 when testing the conan recipe.